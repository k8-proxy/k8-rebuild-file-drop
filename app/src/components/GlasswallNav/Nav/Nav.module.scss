.scrollContainer {
	overflow-y: auto;
	height: 100%;
}

//.navBar {
//	list-style-type: none;
//	margin: 0;
//	padding: 0;
//}

//.nav {
//    height: calc(100% - 19rem);
//    visibility: hidden;
//    opacity: 0;
//    transition: all 0s 0s;

//    &.menuExpanded {
//        opacity: 1;
//        display: block;
//        visibility: visible;
//        transition: visibility 0s 0.3s;
//    }

//}

//.bottom {
//	position: absolute;
//	height: auto;
//	bottom: 5rem;
//	width: 100%;
//}
