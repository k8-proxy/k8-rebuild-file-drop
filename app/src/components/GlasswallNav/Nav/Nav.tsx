import React from "react";

import styles from "./Nav.module.scss";

export interface NavProps { expanded: boolean, bottom: boolean, children?: React.ReactNode }

export type Nav = NavProps;

const Nav = (props: NavProps) => {
    return (
        <nav className={`${styles.nav} ${props.expanded ? styles.menuExpanded : ""} ${props.bottom ? styles.bottom : ""}`}>
            <div className={styles.scrollContainer}>
                <ul>
                    {props.children}
                </ul>
            </div>
        </nav>
    );
};

export { Nav };