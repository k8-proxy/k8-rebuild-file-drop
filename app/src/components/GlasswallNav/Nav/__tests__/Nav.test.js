import React from "react";
import { mount } from "enzyme";
import TestRenderer from "react-test-renderer";
import { Nav } from "../Nav";

const expanded = true;
const bottom = false;

const props = {
    expanded: expanded,
    bottom: bottom
};

test("Nav_Snapshot", () => {
    // Arrange
    const component = TestRenderer.create(<Nav {...props} />);

    // Act
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("Displays_Correct_Props", () => {
    // Arrange
    // Act
    const navComponent = mount(<Nav {...props}/>);

    // Assert
    expect(navComponent.prop("expanded")).toEqual(props.expanded);
    expect(navComponent.prop("clickHandler")).toEqual(props.clickHandler);    
});