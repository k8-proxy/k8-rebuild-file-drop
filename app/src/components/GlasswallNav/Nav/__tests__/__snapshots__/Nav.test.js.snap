// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav_Snapshot 1`] = `
<nav
  className="nav menuExpanded "
>
  <div
    className="scrollContainer"
  >
    <ul />
  </div>
</nav>
`;
